furusiya/model/party/player.py

```diff
@@ -28,14 +37,16 @@ def __init__(self):
         print("You hold your wicked-looking {} at the ready!".format(weapon_name))
 
     def gain_xp(self, amount):
-        self.fighter.xp += amount
+        fighter = self.get_component(Fighter)
+
+        fighter.xp += amount
         # XP doubles every level. 40, 80, 160, ...
         # First level = after four orcs. Yeah, low standards.
         xp_next_level = 2 ** (self.level + 1) * 10
         # DRY ya'ne
-        while self.fighter.xp >= xp_next_level:
+        while fighter.xp >= xp_next_level:
```

Comment:
It strikes me now that maybe the fighter shouldn't have xp. Both values should be defined in the same component (XP and next-level XP). Maybe we need an `gain_experience` component?